### PR TITLE
fix(watermark): can be hidden through the Hide Element feature in the browser

### LINKED ISSUE #45278: Watermark在控制台可通过Hide Element方式隐藏
### Reproduction link

[https://ant.design/components/watermark-cn?theme=happy-work#watermark-demo-image](https://ant.design/components/watermark-cn?theme=happy-work#watermark-demo-image)

### Steps to reproduce

1、开启控制台Elements面板。
2、检查一个水印的base64图片的元素。
3、右键选择Hide Element。
4、水印消失了。

### What is expected?

删除或者修改水印宽高都无法去掉水印，但是隐藏元素可以，期望有人通过控制台操作的时候Hide Element也无法去掉水印

### What is actually happening?

目前很简单就能去掉水印

| Environment | Info |
| --- | --- |
| antd | 5.10.0 |
| React | - |
| System | macOs |
| Browser | Chrome |

---
-

### PR DESCRIPTION
**Pull request by Yuiai01**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close https://github.com/ant-design/ant-design/issues/45278
### 💡 Background and solution

浏览器是通过给元素添加特殊的类来完成 `Hide Element` 的操作，因此在重新生成水印的时候去掉 `class` 属性即可。

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | fix can be hidden watermark through the Hide Element feature in the browser           |
| 🇨🇳 Chinese | 解决可以使用浏览器的 `Hide Element` 来隐藏水印          |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 2263242</samp>

Remove `class` attribute from watermark element in `useWatermark.tsx` to prevent hiding it. This enhances the security of the watermark feature.

### <samp>🤖 Generated by Copilot at 2263242</samp>

* Remove `class` attribute from watermark element to prevent hiding ([link](https://github.com/ant-design/ant-design/pull/45290/files?diff=unified&w=0#diff-3ab28a1f34a8257eb7f273ddccc406fa9bdb28b66dbf50c4b0b7e1260b33c469R44-R46))

### CODE REVIEW

File: components/watermark/useWatermark.tsx [hunk 1]
```diff
@@ -41,6 +41,9 @@ export default function useWatermark(
           backgroundSize: `${Math.floor(markWidth)}px`,
         }),
       );
+      // Prevents using the browser `Hide Element` to hide watermarks
+      watermarkEle.removeAttribute('class');
+
```
Comment (yoyo837) [inline components/watermark/useWatermark.tsx:46]: 如果添加的样式是style呢

Comment (Yuiai01) [reply components/watermark/useWatermark.tsx:46]: style 是生成后固定的，添加新的 style 应该会在重新生成的过程中被刷掉。

Comment (yoyo837) [reply components/watermark/useWatermark.tsx:46]: 层级选择器class呢

Comment (yoyo837) [reply components/watermark/useWatermark.tsx:46]: 比如.xxxx 下这个水印dom

Comment (yoyo837) [reply components/watermark/useWatermark.tsx:46]: 噢，这个没有style优先级高

Comment (yoyo837) [reply components/watermark/useWatermark.tsx:46]: 层级选择器class里加!important呢

Comment (Yuiai01) [reply components/watermark/useWatermark.tsx:46]: > 层级选择器class里加!important呢

这种确实能实现去掉水印的操作，这种就不知道咋防范了。

Comment (Yuiai01) [reply components/watermark/useWatermark.tsx:46]: > 试试给style的样式也加!important

但是这样是不是要加的有点多，visibility、display、width、height等

Comment (yoyo837) [reply components/watermark/useWatermark.tsx:46]: 是的，虽然不好看，但得先满足功能要求吧？水印应该强势和强调自己。让大佬们决断一下看看

Comment (zombieJ) [reply components/watermark/useWatermark.tsx:46]: 小而黑而巧，有意思~

**Review by zombieJ — APPROVED** [review]

File: components/watermark/useWatermark.tsx (hunk 1 above)
Comment (zombieJ) [reply components/watermark/useWatermark.tsx:46]: `visibility` 这个可以加一下，其他的以后有了 issue 再一个个加。

File: components/watermark/useWatermark.tsx [hunk 2]
```diff
@@ -39,6 +44,7 @@ export default function useWatermark(
           ...markStyle,
           backgroundImage: `url('${base64Url}')`,
           backgroundSize: `${Math.floor(markWidth)}px`,
+          ...(EmphasizedStyles as React.CSSProperties),
```
Comment (yoyo837) [inline components/watermark/useWatermark.tsx:47]: markStyle 里面是不是也有 EmphasizedStyles 同名的样式, 加 EmphasizedStyles 的话, 上面的应该去掉.

Comment (Yuiai01) [reply components/watermark/useWatermark.tsx:47]: 嗯嗯，目前 markStyle 里没有 visibility.

Comment (yoyo837) [reply components/watermark/useWatermark.tsx:47]:
```css
display: none:
opacity: 0;
```

如果使用这两个能绕过去吗?

Comment (Yuiai01) [reply components/watermark/useWatermark.tsx:47]: 可以的，有挺多方法，不过按豆酱大佬说的，我这里先仅加了 visibility。

**Review by zombieJ — APPROVED** [review]

**Pull request merged**